src/main/java/com/sublinks/sublinksapi/interceptors/RateLimiter.java

```diff
@@ -0,0 +1,124 @@
+package com.sublinks.sublinksapi.interceptors;
+
+import com.sublinks.sublinksapi.api.lemmy.v3.site.mappers.LocalSiteRateLimitMapper;
+import com.sublinks.sublinksapi.api.lemmy.v3.site.models.LocalSiteRateLimit;
+import com.sublinks.sublinksapi.instance.models.LocalInstanceContext;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.redis.RedisSystemException;
+import org.springframework.data.redis.core.ReactiveRedisTemplate;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.servlet.HandlerInterceptor;
+import reactor.core.Disposable;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.time.Instant;
+
+@Component
+public class RateLimiter implements HandlerInterceptor {
+  private String userName;
```

Comment:
Doesnt this create a side effect? 

If a user logged in => It sets the username => another not logged in user triggers the ratelimits for the last user?